src/main/java/net/sf/jabref/logic/l10n/LocalizationKey.java

```diff
@@ -11,16 +11,18 @@ public LocalizationKey(String key) {
     }
 
     public String getPropertiesKeyUnescaped() {
-        // space, = and : are not allowed in properties file keys
+        // space, #, !, = and : are not allowed in properties file keys
```

Comment:
I agree that the comment is not really clear, still correct, but for
consistency it made sense to add the new escaped characters.